### PR TITLE
Draft - Update TokenizingTextBox usage for PeoplePicker

### PR DESCRIPTION
**Pull request by michael-hawker**
Fixes #

Update the People Picker to use the new TokenizingTextBox

## PR Type
What kind of change does this PR introduce?

- Refactoring

## What is the current behavior?
Uses preview TokenizingTextBox

## What is the new behavior?
Uses new TokenizingTextBox

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](https://github.com/windows-toolkit/Graph-Controls/blob/master/README.md)
- [ ] Sample in sample app has been added / updated (for bug fixes / features)
    - [ ] Icon has been created (if new sample) following the [Thumbnail Style Guide and templates](https://github.com/windows-toolkit/WindowsCommunityToolkit-design-assets)
- [ ] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Header has been added to all new source files (run *build/UpdateHeaders.bat*)
- [ ] Contains **NO** breaking changes

## TODO

- [ ] Update People Picker to inherit from TokenizingTextBox instead of containing one.
- [ ] Update once https://github.com/windows-toolkit/WindowsCommunityToolkit/issues/3289 is fixed.

### CODE REVIEW

**Comment by michael-hawker** [comment]
@azchohfi I had trouble building the WPF/XAML Islands app with this error:
```
2>H:\code\WindowsGraphControls\Samples\XAML Islands\WPF-Core-GraphApp\WPF-Core-GraphApp.csproj : error MSB4236: The SDK 'Microsoft.NET.Sdk.WindowsDesktop' specified could not be found.  H:\code\WindowsGraphControls\Samples\XAML Islands\WPF-Core-GraphApp\WPF-Core-GraphApp.csproj
```

Is there a new change I need to make to the project file?

Also, I noticed that the `Microsoft.SourceLink.Vsts.Git` package is marked deprecated. Should we remove that from here and the main repo?

**Comment by azchohfi** [comment]
Are you sure you have the Desktop dev workload installed on your VS?
About the package, we need to migrate to Microsoft.SourceLink.AzureRepos.Git.

**Comment by michael-hawker** [comment]
@azchohfi I have these workloads installed:
image
Is there a sub-item I need to check?

**Comment by michael-hawker** [comment]
Alright, I think the last bit left here is to be able to fall back on the Users API [as MGT does here](https://github.com/microsoftgraph/microsoft-graph-toolkit/blob/master/src/components/mgt-people-picker/mgt-people-picker.ts#L681-L717).

Then we should be good to go.

@azchohfi I'm going to try the sub-module thing as a test for the sample app tomorrow. Then that limits the crazy NuGet stuff for now. If not I'll fall-back to the weirdness, but will I need an official NuGet reference or could I do it with MyGet still?

File: SampleTest/MainPage.xaml [hunk 1]
```diff
@@ -29,14 +29,14 @@
         <StackPanel Margin="16,48">
             <wgt:PeoplePicker x:Name="PeopleChooser"/>
             <TextBlock Margin="0,8,0,0" FontWeight="Bold">Picked People:</TextBlock>
-            <ItemsControl ItemsSource="{x:Bind PeopleChooser.PickedPeople}"
+            <!--<ItemsControl ItemsSource="{x:Bind PeopleChooser.PickedPeople}"
```
Comment (azchohfi) [inline SampleTest/MainPage.xaml:32]: Why is this commented out?

File: Samples/XAML Islands/UWP-XamlApplication/SamplePage.xaml [hunk 2]
```diff
@@ -15,14 +15,14 @@
         <StackPanel Margin="16,48">
             <wgt:PeoplePicker x:Name="PeopleChooser"/>
             <TextBlock Margin="0,8,0,0" FontWeight="Bold">Picked People:</TextBlock>
-            <ItemsControl ItemsSource="{x:Bind PeopleChooser.PickedPeople}"
+            <!--<ItemsControl ItemsSource="{x:Bind PeopleChooser.PickedPeople}"
```
Comment (azchohfi) [inline Samples/XAML Islands/UWP-XamlApplication/SamplePage.xaml:18]: Why is this commented out?

File: Microsoft.Toolkit.Graph.Controls/Controls/PeoplePicker/PeoplePicker.xaml [hunk 3]
```diff
@@ -2,36 +2,34 @@
     xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation" 
     xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
     xmlns:local="using:Microsoft.Toolkit.Graph.Controls"
-    xmlns:controls="using:Microsoft.Toolkit.Uwp.UI.Controls">
+    xmlns:ex="using:Microsoft.Toolkit.Uwp.UI.Extensions">
+    <ResourceDictionary.MergedDictionaries>
+        <ResourceDictionary Source="ms-appx:///Microsoft.Toolkit.Uwp.UI.Controls/TokenizingTextBox/TokenizingTextBox.xaml" />
```
Comment (azchohfi) [inline Microsoft.Toolkit.Graph.Controls/Controls/PeoplePicker/PeoplePicker.xaml:7]: Why do we need to reference this here? Shouldn't it be automatic?

**Comment by azchohfi** [comment]
You could do it with myget, but I don't think we should. All you have to do is add the myget to the nuget.config file, but we need to remove it before we ship this to nuget.

File: Microsoft.Toolkit.Graph.Controls/Controls/PeoplePicker/PeoplePicker.xaml (hunk 3 above)
Comment (michael-hawker) [reply Microsoft.Toolkit.Graph.Controls/Controls/PeoplePicker/PeoplePicker.xaml:7]: You would think, but nope. ☹

The search for resources only appears to go from our assembly to the generic.xaml. (see wpf behavior (similar) [here](https://docs.microsoft.com/en-us/dotnet/desktop-wpf/fundamentals/xaml-resources-define#static-resource-lookup-behavior) and [here](https://docs.microsoft.com/en-us/dotnet/desktop-wpf/fundamentals/xaml-resources-define#dynamic-resource-lookup-behavior)) - At least that's what I guess is happening.

File: SampleTest/MainPage.xaml (hunk 1 above)
Comment (michael-hawker) [reply SampleTest/MainPage.xaml]: I hadn't updated these yet to the new property name. I also think that exposes a WinUI bug now: https://github.com/microsoft/microsoft-ui-xaml/issues/1286 - need to investigate more.

Comment (michael-hawker) [reply SampleTest/MainPage.xaml]: Ah, no that was trying to use the `TokenSpacing` property on the people picker causing that. I could uncomment this, but it causes some exceptions with types in the generated code unless I create a separate collection I think... I'll look

**Comment by michael-hawker** [comment]
Wondering if we just make the PeoplePicker a behavior and style in the future though that's not as friendly... anyway, going to merge this in and we can sort out changes later. This should be enough for the sample app.

**Pull request merged**